### PR TITLE
Merge `dev` to `feat-fast-sync` 30.12.2021

### PR DESCRIPTION
**Pull request by rafal-ch**
This PR integrates the recent changes from the `dev` (as of 30.12.2021 11:00am CEST) branch into the `feat-fast-sync` branch.

### CODE REVIEW

File: node/src/components/fetcher/metrics.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@ use prometheus::{IntCounter, Registry};
 use crate::unregister_metric;
 
 #[derive(Debug)]
-pub(crate) struct FetcherMetrics {
+pub(crate) struct Metrics {
```
Comment (Fraser999) [inline node/src/components/fetcher/metrics.rs:6]:
```suggestion
pub(super) struct Metrics {
```

**Review by Fraser999 — APPROVED** [review]
One nitpick.

File: node/src/components/fetcher/metrics.rs (hunk 1 above)
Comment (rafal-ch) [reply node/src/components/fetcher/metrics.rs:6]: Actually, on the `feat-fast-sync` the `Fetcher` further exposes the `Metrics` type via the public `ItemFetcher` trait.

I'll see if I can tune the code structure in order not to expose `Metrics` on the crate level.

Comment (rafal-ch) [reply node/src/components/fetcher/metrics.rs:6]: As agreed on the call, we'll merge as-is.

**Pull request merged**